File: symmetric/example5/decrypt.js
const crypto = require('crypto');
const fs = require('fs');

// Читаем ключ из файла
const keyHex = fs.readFileSync('symmetric_key.txt', 'utf-8');
const key = Buffer.from(keyHex, 'hex');

// Читаем зашифрованные данные из файла
const encryptedData = fs
  .readFileSync('encrypted_message.txt', 'utf-8')
  .split('\n');

// Выделяем IV и зашифрованное сообщение
const iv = Buffer.from(encryptedData[0], 'hex');
const encryptedMessage = encryptedData[1];

// Создаем объект для дешифрования
const decipher = crypto.createDecipheriv('aes-256-cbc', key, iv);

// Дешифруем сообщение
let decrypted = decipher.update(encryptedMessage, 'hex', 'utf-8');
decrypted += decipher.final('utf-8');

console.log('Дешифрованное сообщение:', decrypted);

File: symmetric/example5/decrypt.go
package main

import (
	"crypto/aes"
	"crypto/cipher"
	"encoding/hex"
	"io/ioutil"
	"log"
	"strings"
)

func main() {
	// Считываем симметричный ключ и IV из файла
	keyHex, err := ioutil.ReadFile("symmetric_key.txt")
	if err != nil {
		log.Fatal(err)
	}

	key, err := hex.DecodeString(strings.TrimSpace(string(keyHex)))
	if err != nil {
		log.Fatal(err)
	}

	encryptedData, err := ioutil.ReadFile("encrypted_message.txt")
	if err != nil {
		log.Fatal(err)
	}

	// Разделяем строку на IV и зашифрованное сообщение
	parts := strings.Split(strings.TrimSpace(string(encryptedData)), "\n")
	iv, err := hex.DecodeString(parts[0])
	if err != nil {
		log.Fatal(err)
	}

	encryptedMessage, err := hex.DecodeString(parts[1])
	if err != nil {
		log.Fatal(err)
	}

	// Создаем новый блочный шифр AES
	block, err := aes.NewCipher(key)
	if err != nil {
		log.Fatal(err)
	}

	// Создаем объект расшифровщика с режимом CBC
	mode := cipher.NewCBCDecrypter(block, iv)

	// Дешифруем сообщение
	decrypted := make([]byte, len(encryptedMessage))
	mode.CryptBlocks(decrypted, encryptedMessage)

	// Убираем паддинг, если используется PKCS7
	decrypted = PKCS7Unpad(decrypted)

	// Выводим дешифрованное сообщение
	log.Printf("Дешифрованное сообщение: %s", decrypted)
}

func PKCS7Unpad(data []byte) []byte {
    if len(data) == 0 {
        return nil
    }
    pad := int(data[len(data)-1])
    return data[:len(data)-pad]
}
